### ISSUE TITLE
VC++ Unicode Compile error

### ISSUE THREAD

**Issue by dreamnemo** [chriskohlhoff/asio#188]
747 line in win_iocp_socket_service_base.ipp

    if (HMODULE h = **GetModuleHandle**("NTDLL.DLL"))
      ptr = GetProcAddress(h, "NtSetInformationFile");

is a

    if (HMODULE h = **GetModuleHandleA**("NTDLL.DLL"))
      ptr = GetProcAddress(h, "NtSetInformationFile");

**Comment by ned14** [comment]
I hit this problem too, subscribed to when it gets fixed

**Comment by tvkit** [comment]
Here's the diff to build for both x86 & x64

```
diff --git a/asio/include/asio/detail/impl/win_iocp_socket_service_base.ipp b/asio/include/asio/detail/impl/win_iocp_socket_service_base.ipp
index 37e41ec4..758726f6 100644
--- a/asio/include/asio/detail/impl/win_iocp_socket_service_base.ipp
+++ b/asio/include/asio/detail/impl/win_iocp_socket_service_base.ipp
@@ -744,7 +744,7 @@ win_iocp_socket_service_base::get_nt_set_info()
   void* ptr = interlocked_compare_exchange_pointer(&nt_set_info_, 0, 0);
   if (!ptr)
   {
-    if (HMODULE h = GetModuleHandle("NTDLL.DLL"))
+    if (HMODULE h = GetModuleHandleA("NTDLL.DLL"))
       ptr = reinterpret_cast<void*>(GetProcAddress(h, "NtSetInformationFile"));
 
     // On failure, set nt_set_info_ to a special value to indicate that the

```

**Comment by cedral** [comment]
Me too.

**Comment by chriskohlhoff** [comment]
Fixed in 70249b7969aabdeb1279bd7f4c50a63b8b412668.

**Comment by sasmaster** [comment]
He,just got this one in Qt.Thanks for the fix!
